rockcraft.yaml

```diff
@@ -176,29 +176,29 @@ parts:
       - procps
       - openjdk-11-jre-headless
       - python3-setuptools
-    overlay-script: |
-      SPARK_GID=185
-      SPARK_UID=185
 
-      # Create a user in the $CRAFT_OVERLAY chroot
-      groupadd -R $CRAFT_OVERLAY -g ${SPARK_GID} spark
-      useradd -R $CRAFT_OVERLAY -m -r -g spark -u ${SPARK_UID} spark
     override-prime: |
-      SPARK_GID=185
-      SPARK_UID=185
+      SPARK_GID=584792
+      SPARK_UID=584792
 
       craftctl default
-      chown -R ${SPARK_GID}:${SPARK_UID} opt/spark
-      chmod -R 770 opt/spark
+      chmod 755 opt/spark-client/python/bin/*
+
+      chown -R ${SPARK_GID}:${SPARK_UID} etc
+      chmod -R 770 etc
```

Comment:
Is this necessary? seems too relaxed to me. If there are specific nodes in the `/etc` path that spark needs to modify, OK. But globally setting the privilege to `770` violates principles of least privilege. Ditto for changing ownership of the whole subtree.

Reply:
The charm (or the user) needs to change the configuration that are sitting on `/etc/spark` (or related components) as well as write the logs under `/var/log`. So those MUST to be writable. Also note that these settings are applied only to files relevant for this specific rockcraft part. In particular, if I inspect the final image, not everything is owned and permission open:

```
_daemon_@03c75bbdd545:/$ ls -l /var/
total 36
drwxr-xr-x 2 root     root     4096 Apr 18  2022 backups
drwxrwx--- 1 _daemon_ _daemon_ 4096 Aug 31 16:13 cache
drwxr-xr-x 1 root     root     4096 Aug 31 16:13 lib
drwxrwsr-x 2 root     staff    4096 Apr 18  2022 local
lrwxrwxrwx 1 root     root        9 Aug 16 02:03 lock -> /run/lock
drwxrwx--- 1 _daemon_ _daemon_ 4096 Aug 31 16:13 log
drwxrwsr-x 2 root     mail     4096 Aug 16 02:03 mail
drwxr-xr-x 2 root     root     4096 Aug 16 02:03 opt
lrwxrwxrwx 1 root     root        4 Aug 16 02:03 run -> /run
drwxr-xr-x 2 root     root     4096 Aug 16 02:03 spool
drwxrwxrwt 2 root     root     4096 Aug 16 02:06 tmp
```

and 

```
_daemon_@03c75bbdd545:/$ ls -l /etc/
total 400
-rw-r--r-- 1 root     root      3028 Aug 16 02:03 adduser.conf
drwxrwx--- 1 _daemon_ _daemon_  4096 Aug 31 16:13 alternatives
drwxr-xr-x 8 root     root      4096 Aug 16 02:03 apt
-rw-r--r-- 1 root     root      2319 Jan  6  2022 bash.bashrc
-rw-r--r-- 1 root     root       367 Dec 16  2020 bindresvport.blacklist
drwxrwx--- 3 _daemon_ _daemon_  4096 Aug 31 16:13 ca-certificates
-rwxrwx--- 1 _daemon_ _daemon_  5892 Aug 31 16:12 ca-certificates.conf
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 cloud
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 cron.d
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 cron.daily
-rw-r--r-- 1 root     root      2969 Feb 20  2022 debconf.conf
-rw-r--r-- 1 root     root        13 Aug 22  2021 debian_version
drwxrwx--- 1 _daemon_ _daemon_  4096 Aug 31 16:13 default
-rw-r--r-- 1 root     root       604 Sep 15  2018 deluser.conf
drwxr-xr-x 4 root     root      4096 Aug 16 02:06 dpkg
-rw-r--r-- 1 root     root       685 Jan  8  2022 e2scrub.conf
-rw-r--r-- 1 root     root       106 Aug 16 02:03 environment
drwxrwx--- 4 _daemon_ _daemon_  4096 Aug 31 16:13 fonts
-rw-r--r-- 1 root     root        37 Aug 16 02:03 fstab
-rw-r--r-- 1 root     root      2584 Feb  3  2022 gai.conf
-rw-r--r-- 1 root     root       465 Aug 31 16:13 group
-rw-r----- 1 root     shadow     374 Aug 16 02:03 gshadow
drwxr-xr-x 3 root     root      4096 Feb 21  2022 gss
-rw-r--r-- 1 root     root        92 Oct 15  2021 host.conf
-rw-r--r-- 1 root     root        13 Sep  1 10:51 hostname
-rw-r--r-- 1 root     root       174 Sep  1 10:51 hosts
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 init.d
-rwxrwx--- 1 _daemon_ _daemon_  1748 Jan  6  2022 inputrc
-rw-r--r-- 1 root     root        26 Aug  2 13:14 issue
-rw-r--r-- 1 root     root        19 Aug  2 13:14 issue.net
drwxrwx--- 5 _daemon_ _daemon_  4096 Aug 31 16:13 java-11-openjdk
drwxr-xr-x 3 root     root      4096 Aug 16 02:03 kernel
-rwxrwx--- 1 _daemon_ _daemon_  2786 Aug 31 16:13 krb5.conf
-rwxrwx--- 1 _daemon_ _daemon_  9471 Aug 31 16:13 ld.so.cache
-rw-r--r-- 1 root     root        34 Dec 16  2020 ld.so.conf
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 ld.so.conf.d
-rw-r--r-- 1 root     root       267 Oct 15  2021 legal
-rw-r--r-- 1 root     root       191 Mar 17  2022 libaudit.conf
-rw-r--r-- 1 root     root     10734 Nov 11  2021 login.defs
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 logrotate.d
-rw-r--r-- 1 root     root       104 Aug  2 13:13 lsb-release
-rw-r--r-- 1 root     root         0 Aug 16 02:06 machine-id
-rwxrwx--- 1 _daemon_ _daemon_ 72029 Mar 21  2022 mime.types
-rw-r--r-- 1 root     root       744 Jan  8  2022 mke2fs.conf
lrwxrwxrwx 1 root     root        12 Sep  1 10:51 mtab -> /proc/mounts
-rw-r--r-- 1 root     root       767 Mar 24  2022 netconfig
-rw-r--r-- 1 root     root        91 Oct 15  2021 networks
-rw-r--r-- 1 root     root       494 Dec 16  2020 nsswitch.conf
drwxr-xr-x 2 root     root      4096 Aug 16 02:03 opt
lrwxrwxrwx 1 root     root        21 Aug  2 13:14 os-release -> ../usr/lib/os-release
-rw-r--r-- 1 root     root       552 Aug 12  2020 pam.conf
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 pam.d
-rw-r--r-- 1 root     root       987 Aug 31 16:13 passwd
-rw-r--r-- 1 root     root       582 Oct 15  2021 profile
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 profile.d
drwxrwx--- 2 _daemon_ _daemon_  4096 Aug 31 16:13 python3
drwxrwx--- 2 _daemon_ _daemon_  4096 Aug 31 16:13 python3.10
drwxr-xr-x 2 root     root      4096 Feb 15  2022 rc0.d
drwxr-xr-x 2 root     root      4096 Feb 15  2022 rc1.d
drwxr-xr-x 2 root     root      4096 Feb 15  2022 rc2.d
drwxr-xr-x 2 root     root      4096 Feb 15  2022 rc3.d
drwxr-xr-x 2 root     root      4096 Feb 15  2022 rc4.d
drwxr-xr-x 2 root     root      4096 Feb 15  2022 rc5.d
drwxr-xr-x 2 root     root      4096 Feb 15  2022 rc6.d
drwxr-xr-x 2 root     root      4096 Aug 16 02:03 rcS.d
-rw-r--r-- 1 root     root       789 Sep  1 10:51 resolv.conf
lrwxrwxrwx 1 root     root        13 Feb 15  2023 rmt -> /usr/sbin/rmt
drwxr-xr-x 4 root     root      4096 Aug 16 02:06 security
drwxr-xr-x 2 root     root      4096 Aug 16 02:05 selinux
-rw-r--r-- 1 root     root       524 Aug 31 16:13 shadow
-rw-r--r-- 1 root     root       128 Aug 16 02:03 shells
drwxr-xr-x 2 root     root      4096 Aug 16 02:03 skel
drwxrwx--- 3 _daemon_ _daemon_  4096 Aug 31 16:13 spark
drwxrwx--- 3 _daemon_ _daemon_  4096 Aug 31 16:13 spark8t
drwxrwx--- 3 _daemon_ _daemon_  4096 Aug 31 16:13 ssl
-rw-r--r-- 1 root     root         0 Aug 16 02:03 subgid
-rw-r--r-- 1 root     root         0 Aug 16 02:03 subuid
-rw-r--r-- 1 root     root      2355 Feb 25  2022 sysctl.conf
drwxr-xr-x 2 root     root      4096 Aug 16 02:05 sysctl.d
drwxr-xr-x 4 root     root      4096 Feb 15  2022 systemd
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 terminfo
-rwxrwx--- 1 _daemon_ _daemon_  1260 Jun 16  2020 ucf.conf
drwxr-xr-x 2 root     root      4096 Aug 16 02:06 update-motd.d
-rw-r--r-- 1 root     root       681 Mar 23  2022 xattr.conf
```

Reply:
One thought I would have is whether we should actually reduce the group permission, like going  to `750` or so